### ISSUE TITLE
Ionic radio buttons - label intercepts pointer events

### ISSUE THREAD

**Issue by alxamarin** [microsoft/playwright#8813]
**Context:**
- Playwright Version: 1.14.1
- Operating System: Windows 10
- Node.js version: 14.17.5
- Browser: Chromium headless

**Code Snippet**

```typescript
    // await this.page.check(`ion-radio[value='${userRole.toLowerCase()}'] >> input[type=radio]`);
    const roleEl = await this.page.$(`ion-radio[value='${userRole.toLowerCase()}'] >> input[type=radio]`);
    // const roleEl = await this.page.$(`ion-radio[value='${userRole.toLowerCase()}'] >> label`); // input[type=radio]
    await roleEl.check();
```

**Describe the bug**

I test Ionic app which uses shadow DOM. And I need to check one of many radio buttons. But when I do this using the code above I get this output:
>elementHandle.check: Timeout 3000ms exceeded.
    =========================== logs ===========================
    attempting click action
      waiting for element to be visible, enabled and stable
      element is visible, enabled and stable
      scrolling into view if needed
      done scrolling
      checking that element receives pointer events at (905.37,492)
      <label for="ion-rb-1"></label> intercepts pointer events

and then it retries. Please look at this:
![ionic shadow dom radio button problem](https://inus.azureedge.net/pub/ionic-radio-playwright-problem.jpg)

If I use the selector which points exactly at `input[type=radio]` I get the message:
><label for="ion-rb-1"></label> intercepts pointer events

At end I gave up as I couldn't find any way of checking this radio button.

**Comment by dgozman** [comment]
I tried to repro on this page: https://ionicframework.com/docs/api/radio. There is indeed a label over the actual radio input, so the error is expected.

However, we should make it work when you point to `ion-radio` instead because it has a proper `role=radio`:
```js
await this.page.check(`ion-radio[value='${userRole.toLowerCase()}']`);
```

**Comment by alxamarin** [comment]
but it doesn't since it's not an radio button (not input[type=radio]). Here's the hack that works:
```
    const radioSelector = `#id-ion-radio-${userRole.toLowerCase()}`;
    await this.page.click(radioSelector);
    await this.page.click(`${radioSelector} label`);
    await this.page.click(radioSelector);
    await this.page.click(`${radioSelector} label`);
```
and this is kind of ugly.

**Comment by dgozman** [comment]
@alxamarin It does not work for `ion-radio` currently, but that's a bug. We should allow that because it has proper `role=radio` attribute. Should be fixed in the next release with #8821.

**Comment by alxamarin** [comment]
@dgozman, thank you!
